src/ImageSharp/Metadata/Profiles/IPTC/IptcProfile.cs

```diff
@@ -264,34 +268,34 @@ private void Initialize()
 
             this.values = new Collection<IptcValue>();
 
-            if (this.Data == null || this.Data[0] != 0x1c)
+            if (this.Data == null || this.Data[0] != IptcTagMarkerByte)
             {
                 return;
             }
 
-            int i = 0;
-            while (i + 4 < this.Data.Length)
+            int offset = 0;
+            while (offset + 4 < this.Data.Length)
```

Comment:
```suggestion
            while (offset < this.Data.Length - 4)
```
So the "loop variable" `offset` can be kept in a register, avoiding the repeated addition. Or in other words `this.Data.Length - 4` is constant for the loop, thus will be held in a register.

Also see comment about overflow.